services/repository/fork.go

```diff
@@ -70,14 +71,18 @@ func ForkRepository(ctx context.Context, doer, owner *user_model.User, opts Fork
 		}
 	}
 
+	defaultBranch := opts.BaseRepo.DefaultBranch
```

Comment:
If opts.SingleBranch, then it should fork with all branches but not default branch?